Failed to execute "mvn install" at branch 2.x

**Describe the bug**

```java
gitpod /workspace/springdoc-openapi (bug/fix-build-error) $ mvn install -DskipTests
Picked up JAVA_TOOL_OPTIONS:  -Xmx3435m
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] Spring openapi documentation                                       [pom]
[INFO] springdoc-openapi-starter-common                                   [jar]
[INFO] springdoc-openapi-starter-webmvc-api                               [jar]
[INFO] springdoc-openapi-starter-webflux-api                              [jar]
[INFO] springdoc-openapi-starter-webmvc-ui                                [jar]
[INFO] springdoc-openapi-starter-webflux-ui                               [jar]
[INFO] springdoc-openapi-tests                                            [pom]
[INFO] springdoc-openapi-security-tests                                   [jar]
[INFO] springdoc-openapi-groovy-tests                                     [jar]
[INFO] springdoc-openapi-javadoc-tests                                    [jar]
[INFO] springdoc-openapi-function-webmvc-tests                            [jar]
[INFO] springdoc-openapi-function-webflux-tests                           [jar]
[INFO] springdoc-openapi-actuator-webflux-tests                           [jar]
[INFO] springdoc-openapi-actuator-webmvc-tests                            [jar]
[INFO] springdoc-openapi-kotlin-tests                                     [jar]
[INFO] springdoc-openapi-native-tests                                     [jar]
[INFO] springdoc-openapi-hateoas-tests                                    [jar]
[INFO] springdoc-openapi-data-rest-tests                                  [jar]
[INFO] 
[INFO] ------------------< org.springdoc:springdoc-openapi >-------------------
[INFO] Building Spring openapi documentation 2.0.0-M4-SNAPSHOT           [1/18]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- maven-install-plugin:2.5.2:install (default-install) @ springdoc-openapi ---
[INFO] Installing /workspace/springdoc-openapi/pom.xml to /workspace/m2-repository/org/springdoc/springdoc-openapi/2.0.0-M4-SNAPSHOT/springdoc-openapi-2.0.0-M4-SNAPSHOT.pom
[INFO] 
[INFO] -----------< org.springdoc:springdoc-openapi-starter-common >-----------
[INFO] Building springdoc-openapi-starter-common 2.0.0-M4-SNAPSHOT       [2/18]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Spring openapi documentation 2.0.0-M4-SNAPSHOT:
[INFO] 
[INFO] Spring openapi documentation ....................... SUCCESS [  0.236 s]
[INFO] springdoc-openapi-starter-common ................... FAILURE [  0.468 s]
[INFO] springdoc-openapi-starter-webmvc-api ............... SKIPPED
[INFO] springdoc-openapi-starter-webflux-api .............. SKIPPED
[INFO] springdoc-openapi-starter-webmvc-ui ................ SKIPPED
[INFO] springdoc-openapi-starter-webflux-ui ............... SKIPPED
[INFO] springdoc-openapi-tests ............................ SKIPPED
[INFO] springdoc-openapi-security-tests ................... SKIPPED
[INFO] springdoc-openapi-groovy-tests ..................... SKIPPED
[INFO] springdoc-openapi-javadoc-tests .................... SKIPPED
[INFO] springdoc-openapi-function-webmvc-tests ............ SKIPPED
[INFO] springdoc-openapi-function-webflux-tests ........... SKIPPED
[INFO] springdoc-openapi-actuator-webflux-tests ........... SKIPPED
[INFO] springdoc-openapi-actuator-webmvc-tests ............ SKIPPED
[INFO] springdoc-openapi-kotlin-tests ..................... SKIPPED
[INFO] springdoc-openapi-native-tests ..................... SKIPPED
[INFO] springdoc-openapi-hateoas-tests .................... SKIPPED
[INFO] springdoc-openapi-data-rest-tests .................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.158 s
[INFO] Finished at: 2022-07-22T09:30:04Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project springdoc-openapi-starter-common: Could not resolve dependencies for project org.springdoc:springdoc-openapi-starter-common:jar:2.0.0-M4-SNAPSHOT: Failed to collect dependencies at org.springframework.cloud:spring-cloud-function-web:jar:4.0.0-M2: Failed to read artifact descriptor for org.springframework.cloud:spring-cloud-function-web:jar:4.0.0-M2: io.micrometer:micrometer-bom:pom:2.0.0-M3 was not found in https://repo.spring.io/release during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of spring-release has elapsed or updates are forced -> [Help 1]
```

**To Reproduce**
Steps to reproduce the behavior:
- Checkout from `v2.0.0-M3`
- Execute `mvn install -DskipTests`
- See the error

**Expected behavior**

```java
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Spring openapi documentation 2.0.0-M4-SNAPSHOT:
[INFO] 
[INFO] Spring openapi documentation ....................... SUCCESS [  0.224 s]
[INFO] springdoc-openapi-starter-common ................... SUCCESS [  1.297 s]
[INFO] springdoc-openapi-starter-webmvc-api ............... SUCCESS [  0.183 s]
[INFO] springdoc-openapi-starter-webflux-api .............. SUCCESS [  0.090 s]
[INFO] springdoc-openapi-starter-webmvc-ui ................ SUCCESS [  0.104 s]
[INFO] springdoc-openapi-starter-webflux-ui ............... SUCCESS [  0.120 s]
[INFO] springdoc-openapi-tests ............................ SUCCESS [  0.006 s]
[INFO] springdoc-openapi-security-tests ................... SUCCESS [  0.048 s]
[INFO] springdoc-openapi-groovy-tests ..................... SUCCESS [  2.548 s]
[INFO] springdoc-openapi-javadoc-tests .................... SUCCESS [  0.148 s]
[INFO] springdoc-openapi-function-webmvc-tests ............ SUCCESS [  0.038 s]
[INFO] springdoc-openapi-function-webflux-tests ........... SUCCESS [  0.063 s]
[INFO] springdoc-openapi-actuator-webflux-tests ........... SUCCESS [  0.033 s]
[INFO] springdoc-openapi-actuator-webmvc-tests ............ SUCCESS [  0.071 s]
[INFO] springdoc-openapi-kotlin-tests ..................... SUCCESS [  9.282 s]
[INFO] springdoc-openapi-native-tests ..................... SUCCESS [  0.058 s]
[INFO] springdoc-openapi-hateoas-tests .................... SUCCESS [  0.116 s]
[INFO] springdoc-openapi-data-rest-tests .................. SUCCESS [  0.098 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  14.877 s
[INFO] Finished at: 2022-07-22T09:33:07Z
[INFO] ------------------------------------------------------------------------
```

**Additional context**

I confirmed that We could resolve the issue after upgrading `spring-cloud-function` to `4.0.0-M3`.

BTW, I'm willing to fire a PR against this issue.

/assign

--- a/pom.xml
+++ b/pom.xml
@@ -95,7 +95,7 @@
 		<jjwt.version>0.9.1</jjwt.version>
 		<spring-native.version>0.11.5</spring-native.version>
 		<therapi-runtime-javadoc.version>0.13.0</therapi-runtime-javadoc.version>
-		<spring-cloud-function.version>4.0.0-M2</spring-cloud-function.version>
+		<spring-cloud-function.version>4.0.0-M3</spring-cloud-function.version>
 	</properties>
 
 	<dependencyManagement>

